spdx-utils/src/main/antlr/com/here/ort/spdx/SpdxExpression.g4

```diff
@@ -83,7 +83,8 @@ OPEN  : '(' ;
 CLOSE : ')' ;
 PLUS  : '+' ;
 
-LICENSEREFERENCE : ('DocumentRef-' IDSTRING ':')? 'LicenseRef-' IDSTRING ;
-IDSTRING         : (ALPHA | DIGIT)(ALPHA | DIGIT | '-' | '.')* ;
+IDSTRING          : (ALPHA | DIGIT)(ALPHA | DIGIT | '-' | '.')* ;
+LICENSE_EXCEPTION : (IDSTRING '-exception' | IDSTRING '-exception-' IDSTRING) ;
```

Comment:
It's true that all "official" SDPX exceptions contain "-exception-", and that what comes after `WITH` needs to be in that list, but it does not specifically define that "-exception-" needs to be contained. For our lenient implementation that shall be able to parse also expressions with typos or non-SPDX IDs I would not make that requirement.
Is there any specific problem you want so solve? Otherwise I would not make the grammar more complex than it needs to be.

Reply:
The specific issue I wanted to solve was the bogus `line 1:8 no viable alternative at input 'licensea'` console output I see when running `PackageCurationTest`. I noticed the ambiguity between `licenseIdExpression` and `licenseExceptionExpression` while looking for the cause, and wanted to address it along the way. In any case, this change does not fix the "no viable alternative at input" issue.